### ISSUE TITLE
False positive for UnusedPrivateMethod when method is called on `new static()`

### ISSUE THREAD

**Issue by oliverklee** [phpmd/phpmd#900]
- PHPMD version: 2.10.1snapshot202105111723
- PHP Version: 7.4.20
- Installation type: phar file
- Operating System / Distribution & Version: Ubuntu 21.04

### Current Behavior
Error messages, including this one:
```
/home/klee/src/php/emogrifier/src/HtmlProcessor/AbstractHtmlProcessor.php:119   Avoid unused private methods such as 'setHtml'.
```

This is the affected piece of code:
```php
    public static function fromHtml(string $unprocessedHtml): self
    {
        if ($unprocessedHtml === '') {
            throw new \InvalidArgumentException('The provided HTML must not be empty.', 1515763647);
        }

        $instance = new static();
        $instance->setHtml($unprocessedHtml);

        return $instance;
    }
```

### Expected Behavior
No PHPMD warnings.

### Steps To Reproduce:
1. git clone git@github.com:MyIntervals/emogrifier.git
2. cd emogrifier
3. composer ci:php:md

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] I have added every step to reproduce the bug.
* [x] If possible I added relevant code examples.
* [x] This issue is about 1 bug and nothing more.
* [x] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by kylekatarnls** [comment]
Hello, thanks for your report.

I guess this is a reduced version and the real code contains 2 methods right?

Else I don't see how `fromHtml()` can end to be called here.

**Comment by oliverklee** [comment]
Hi @kylekatarnls, thanks for the quick reply! Yes, the real code indeed contains more methods, including the `setHtml` method. The public static method `fromHtml`, on the other hand, is part of our library's API, and not called from our own code (except from the unit tests).

The complete code is available at https://github.com/MyIntervals/emogrifier (as mentioned in the steps for reproducing the problem).

**Comment by kylekatarnls** [comment]
Progress on this issue will be tracked in #558
